lab.R

```diff
@@ -139,18 +164,23 @@ levels(bad_factor)
 
 #' To fix this, we need to set the order of the levels manually.  First, edit the next line, so that the levels correspond to the list in the documentation, with "Very Poor" as the lowest (first) value and "Very Excellent" as the highest (<http://jse.amstat.org/v19n3/decock/DataDocumentation.txt>, search for "Overall Cond").  
 
-condition_levels = c('Above_Average', 'Very_Poor', 'Below_Average')
+
+# "Above_Average" "Average"       "Below_Average" "Excellent"     "Fair"          "Good"          "Poor"          "Very_Good"     "Very_Poor"
+
+condition_levels = c("Very_Poor", "Poor", "Fair", "Below_Average", "Average", "Above_Average", "Good", "Very_Good", "Excellent")
 
 #' 4. To fix the levels, we use the `levels = ` argument in `factor()`.  Modify the code for `bad_factor` above, assigning the result to `good_factor`.  Confirm that the levels are in the right order. 
 
-# good_factor = ???
+good_factor = factor(dataf$Overall_Cond, levels = condition_levels)
 
 #' 5. Finally we want this factor to be in our analysis dataframe.  *Normally, to preserve immutability*, I would do this right after reading in the CSV (using a pipe).  We'll talk more about this when we work on cleaning data.  For this lab, just do it here.  
 #' 
 #' Using `mutate()`, add a variable `overall_cond_fct` to `dataf`, that represents overall condition as a factor with the levels in the correct order. 
 
 # dataf = ???
 
+dataf %>% 
+    mutate(overall_cond_fct = good_factor)
```

Comment:
You need to assign the result to a variable (in this case `dataf`)
```suggestion
dataf = dataf %>% 
    mutate(overall_cond_fct = good_factor)
```

Reply:
It looks like most of the errors further down happen because you didn't assign this back to `dataf`.